src/plots/cartesian/type_defaults.js

```diff
@@ -91,6 +91,7 @@ function setAutoType(ax, data) {
         }
     }
     else {
+        if(axLetter === 'z') opts.noMultiCategory = true;
```

Comment:
That works. But `opts.noMultiCategory` should already be `true` from 

https://github.com/plotly/plotly.js/blob/9beb88d02ac0a0566238e1038159c75408476069/src/plots/cartesian/type_defaults.js#L60

I suspect `d0` in this scope is `fullData[0]` when it should be the first trace on the gl3d subplot.

I hope that makes sense :smile:

Reply:
You are right. It seems to be the first trace on fullData which is not invisible.
The issue is not only affected 3d traces. The example below also fails on master:
<pre>
Plotly.newPlot(gd, {
    "data": [
      {
        "type": "heatmap",
        "x": [0, 1, 2],
        "y": [0, 1, 2],
        "z": [
          [0, 1, 0],
          [1, 0, 1],
          [0, 1, 0]
        ]
      },
      {
        "type": "scattergl",
        "x": [0, 1, 2],
        "y": [0, 1, 2]
      }
    ],
    "layout": {
        "title": "scattergl plot on top of 2d heatmap",
        "width": 600,
        "height": 400
    }
});
</pre>

Reply:
@etpinard The 2d case was mentioned above failed on v.1.43.0; but works on master. So I would fetch master changes and fix this for the 3D case as you mentioned.